libnetwork/drivers/remote/driver.go

```diff
@@ -199,6 +199,16 @@ func (d *driver) CreateEndpoint(nid, eid string, ifInfo driverapi.InterfaceInfo,
 		return err
 	}
 
+	defer func() {
+		if retErr != nil {
+			if err := d.DeleteEndpoint(nid, eid); err != nil {
+				retErr = fmt.Errorf("%w; failed to roll back: %w", err, retErr)
+			} else {
+				retErr = fmt.Errorf("%w; rolled back", retErr)
```

Comment:
Honestly (but it was already there, so maybe we need to keep it), I'm tempted to say that for this case we don't have to add the `rolled back` part. Effectively, we tried to create something and it failed. The "rollback" is an implementation detail that's not really relevant (it was partially created, and it's now gone, so .... basically it never happened).

Reply:
Yeah, I had the same thought and then prefered to keep it. I think it can still be useful to know that a specific operation has been properly rolled back. Maybe a log would be better?